meson.build

```diff
@@ -0,0 +1,467 @@
+project('gerbera', 'c', 'cpp',
+  version : '1.9.1',
+  default_options : [ 'c_std=gnu11', 'cpp_std=c++17', 'warning_level=1' ],
+)
+
+cpp = meson.get_compiler('cpp')
+if cpp.get_id() == 'gcc' and cpp.version().version_compare('<8')
+  error('Your GCC version is too old.  You need at least version 8')
+elif cpp.get_id() == 'clang' and cpp.version().version_compare('<7')
+  error('Your clang version is too old.  You need at least version 7.')
+endif
+
+add_global_arguments('-DGERBERA_VERSION="@0@"'.format(meson.project_version()), language : 'cpp')
+add_global_arguments('-DPACKAGE_DATADIR="@0@/@1@/@2@"'.format(get_option('prefix'), get_option('datadir'), meson.project_name()), language : 'cpp')
+
+# This makes assumptions that the libcpp is the GNU one on Linux systems
+if cpp.version().version_compare('>=9')
+  fs_dep = declare_dependency()
+elif host_machine.system() == 'linux'
+  fs_dep = cpp.find_library('stdc++fs')
+elif cpp.get_id() == 'clang'
+  fs_dep = cpp.find_library('c++fs')
+endif
+
+fmt_dep = dependency('fmt', version : '>=6.1.2')
+pugixml_dep = dependency('pugixml')
+spdlog_dep = dependency('spdlog', version : '>=1.5.0')
+sqlite3_dep = dependency('sqlite3', version : '>=3.7.11')
+
+iconv_dep = cpp.find_library('iconv', required : false)
+if not iconv_dep.found()
+  if cpp.has_function('iconv')
+    iconv_dep = declare_dependency()
+  else
+    error('iconv implementation is needed')
+  endif
+endif
+
+if cpp.links('''
+  #include <stdint.h>
+  int main(void) {
+    uint32_t x32 = 0;
+    uint64_t x64 = 0;
+    __atomic_load_n(&x32, __ATOMIC_SEQ_CST);
+    __atomic_load_n(&x64, __ATOMIC_SEQ_CST);
+    return 0;
+    }''', name : 'built-in atomics')
+  libatomic_dep = declare_dependency()
+else
+  libatomic_dep = cpp.find_library('atomic')
+endif
+
+libgerbera_args = []
+
+if (cpp.has_function('nl_langinfo'))
+  libgerbera_args += '-DHAVE_NL_LANGINFO'
+endif
+
+if (cpp.has_function('setlocale'))
+  libgerbera_args += '-DHAVE_SETLOCALE'
+endif
+
+uuid_dep = dependency('uuid', required : false)
+if not uuid_dep.found()
+  if host_machine.system() == 'linux'
+    error('libuuid is required')
+  else
+    libgerbera_args += '-DBSD_NATIVE_UUID'
+  endif
+endif
+
+# Link to libnsl (Network services library) if it exists. This is something you need on Solaris/OmniOS/Joyent
+nsl_dep = dependency('nsl', required : false)
+# Link to the socket library if it exists. This is something you need on Solaris/OmniOS/Joyent
+socket_dep = dependency('socket', required : false)
+
+if host_machine.system() == 'solaris'
+  libgerbera_args += '-DSOLARIS'
+elif host_machine.system() != 'linux'
+  libgerbera_args += '-DBSD'
+endif
+
+if get_option('buildtype') != 'debug'
```

Comment:
```
get_option('debug')
```

buildtype is a combo of optimization+debug.

Reply:
Could you elaborate on this?

My initial attempt at adding WITH_DEBUG failed as debug seems to be a reserved option.

Reply:
https://mesonbuild.com/Builtin-options.html#core-options

"buildtype" is a core option, "optimization" and "debug" are slightly newer core options which bidirectionally map to buildtype. Instead of checking whether the *buildtype* option is "debug" (or debugoptimized! which you do not handle), you can check whether the "debug" option is boolean true.

See https://mesonbuild.com/Release-notes-for-0-48-0.html#toggles-for-build-type-optimization-and-vcrt-type

You cannot add a "debug" option to meson_options.txt if that is what you were trying.